xml/schema/types/editors.xsd

```diff
@@ -1159,7 +1159,7 @@
       <xs:attribute name="ident" type="beanNameType" use="required">
         <xs:annotation><xs:documentation>Identify this element, for reference in connect* attributes elsewhere</xs:documentation></xs:annotation>
       </xs:attribute>
-      <xs:attribute name="type" type="xs:int" use="required"></xs:attribute>
+      <xs:attribute name="type" type="xs:string" use="required"></xs:attribute>
```

Comment:
Our schema are meant to check contents (whenever possible).  To validate strings for the enum values, please enumerate them with `<xs:restriction>` and `<xs:restriction>`.  Also, see the [JMRI page on XML schema](https://www.jmri.org/help/en/html/doc/Technical/XmlSchema.shtml) for how to update the version number.  Thanks!

Reply:
If the schema is changed to "xs:enumeration" then backwards compatibility is lost. 
For now "xs:string" allows both the previous integers (since they're actually saved as strings) AND the new enumeration strings to load.
In the first major release after the next one backwards compatibility with integers can be removed; the xml integer value code removed and the schema updated to "xs:enumeration".

Reply:
Ok, I think I've got it… copy xml/schema/types/editors.xsd to editors-4-19-5.xsd and add TurnoutTypeEnumType support:
`

      <xs:attribute name="type" type="TurnoutTypeEnumType" use="required"></xs:attribute>
    .
    .
    .
    <xs:simpleType name="TurnoutTypeEnumType">
      <xs:restriction base="xs:string">
        <xs:pattern value="(NONE|0)"></xs:pattern>
        <xs:pattern value="(RH_TURNOUT|1)"></xs:pattern>
        <xs:pattern value="(LH_TURNOUT|2)"></xs:pattern>
        <xs:pattern value="(WYE_TURNOUT|3)"></xs:pattern>
        <xs:pattern value="(DOUBLE_XOVER|4)"></xs:pattern>
        <xs:pattern value="(RH_XOVER|5)"></xs:pattern>
        <xs:pattern value="(LH_XOVER|6)"></xs:pattern>
        <xs:pattern value="(SINGLE_SLIP|7)"></xs:pattern>
        <xs:pattern value="(DOUBLE_SLIP|8)"></xs:pattern>
      </xs:restriction>
    </xs:simpleType>
`
Then copy layout-4-19-2.xsd to layout-4-19-5.xsd and change it to reference the new <editors-4-19-5.xsd> file above:

`  <xs:include schemaLocation="http://jmri.org/xml/schema/types/editors-4-19-5.xsd"/>`

Then change java/src/jmri/configurexml/ConfigXmlManager.java to use the new schema:

`    static final public String schemaVersion = "-4-19-5";`

Then at some point in the future the integer values can be removed.

Reply:
Re backward compatibility:

```
        <xs:restriction base="xs:token" >
          <xs:enumeration value="1"/>
          <xs:enumeration value="foo"/>
          <xs:enumeration value="2"/>
          <xs:enumeration value="bar"/>
          <xs:enumeration value="3"/>
          <xs:enumeration value="biff"/>
        </xs:restriction base="xs:token" >
```
which also validates more than just `xs:int`

Reply:
xs:int is too broad since [4-9] above aren't valid values.
Any problems with using TurnoutTypeEnumType like I did above?

Reply:
@geowar1 , @bobjacobsen isn’t saying to use xs:int.  He is saying to use xs:token.

Reply:
Any advantage to that over the TurnoutTypeEnumType xs: simpleType?

Reply:
> Any advantage to that over the TurnoutTypeEnumType xs: simpleType?

This is for the restriction, not for the TurnoutTyoeEnumType you defined.

The restriction, as specified with the enumeration, is somewhat simpler than the restriction specified with pattern matching.